stable_baselines/sac/policies.py

```diff
@@ -0,0 +1,370 @@
+        :param mask: ([float]) The last masks (used in recurrent policies)
+        :param deterministic: (bool) Whether or not to return deterministic actions.
+        :return: ([float]) actions
+        """
+        raise NotImplementedError
+
+    def proba_step(self, obs, state=None, mask=None):
+        """
+        Returns the action probability for a single step
+
+        :param obs: ([float] or [int]) The current observation of the environment
+        :param state: ([float]) The last states (used in recurrent policies)
+        :param mask: ([float]) The last masks (used in recurrent policies)
+        :return: ([float]) the action probability
+        """
+        raise NotImplementedError
+
+
+class FeedForwardPolicy(SACPolicy):
+    """
+    Policy object that implements a DDPG-like actor critic, using a feed forward neural network.
+
+    :param sess: (TensorFlow session) The current TensorFlow session
+    :param ob_space: (Gym Space) The observation space of the environment
+    :param ac_space: (Gym Space) The action space of the environment
+    :param n_env: (int) The number of environments to run
+    :param n_steps: (int) The number of steps to run for each environment
+    :param n_batch: (int) The number of batch to run (n_envs * n_steps)
+    :param reuse: (bool) If the policy is reusable or not
+    :param layers: ([int]) The size of the Neural network for the policy (if None, default to [64, 64])
+    :param cnn_extractor: (function (TensorFlow Tensor, ``**kwargs``): (TensorFlow Tensor)) the CNN feature extraction
+    :param feature_extraction: (str) The feature extraction type ("cnn" or "mlp")
+    :param layer_norm: (bool) enable layer normalisation
+    :param reg_weight: (float) Regularization loss weight for the policy parameters
+    :param kwargs: (dict) Extra keyword arguments for the nature CNN feature extraction
+    """
+
+    def __init__(self, sess, ob_space, ac_space, n_env=1, n_steps=1, n_batch=None, reuse=False, layers=None,
+                 cnn_extractor=nature_cnn, feature_extraction="cnn", reg_weight=0.0,
+                 layer_norm=False, **kwargs):
+        super(FeedForwardPolicy, self).__init__(sess, ob_space, ac_space, n_env, n_steps, n_batch,
+                                                reuse=reuse, scale=(feature_extraction == "cnn"))
+        self.layer_norm = layer_norm
+        self.feature_extraction = feature_extraction
+        self.cnn_kwargs = kwargs
+        self.cnn_extractor = cnn_extractor
+        self.reuse = reuse
+        if layers is None:
+            layers = [64, 64]
+        self.layers = layers
+        self.reg_loss = None
+        self.reg_weight = reg_weight
+        self.entropy = None
+
+        assert len(layers) >= 1, "Error: must have at least one hidden layer for the policy."
+
+        self.activ_fn = tf.nn.relu
+        # TODO: check perf with tanh instead of relu (original paper)
```

Comment:
add as a hyperparam, and remove TODO?

Reply:
the ReLU yields good perf vs tanh (I'm currently testing with several bullets envs) and I tuned the params for ReLU, so I removed it.